ceph-dev-pipeline/build/Jenkinsfile

```diff
@@ -120,6 +154,73 @@ pipeline {
+              script {
+                def props = readProperties file: "${WORKSPACE}/dist/other_envvars"
+                for (p in props) {
+                  env."${p.key}" = p.value
+                }
+              }
+              sh '''#!/bin/bash
+                set -ex
+                cd dist
+                mkdir ceph
+                tar --strip-components=1 -C ceph -xjf ceph-$VERSION.tar.bz2
+              '''
+            }
+          }
+          stage("check for built packages") {
+            when {
+              environment name: 'THROWAWAY', value: 'false'
+              environment name: 'CI_COMPILE', value: 'true'
+            }
+            environment {
+              CHACRACTL_KEY = credentials('chacractl-key')
+              SHAMAN_API_KEY = credentials('shaman-api-key')
+            }
+            steps {
+              script {
+                sh './scripts/setup_chacractl.sh'
+                def chacra_url = sh(
+                  script: '''grep url ~/.chacractl | cut -d'"' -f2''',
+                  returnStdout: true,
+                ).trim()
+                def os = get_os_info(env.DIST)
+                def chacra_endpoint = "ceph/${env.BRANCH}/${env.SHA1}/${os.name}/${os.version_name}/${env.ARCH}/flavors/${env.FLAVOR}/"
+                def chacractl_rc = sh(
+                  script: "$WORKSPACE/.venv/bin/chacractl exists binaries/${chacra_endpoint}",
+                  returnStatus: true,
+                )
+                if ( chacractl_rc == 0 && env.FORCE != "true" ) {
+                  println("Skipping compilation since chacra already has artifacts. To override, use THROWAWAY=true (to skip this check) or FORCE=true (to re-upload artifacts).")
+                  env.CI_COMPILE = "false"
+                }
+              }
+            }
+          }
+          stage("builder container") {
+            environment {
+              CONTAINER_REPO_CREDS = credentials('quay-ceph-io-ceph-ci')
+            }
+            when {
+              environment name: 'CI_COMPILE', value: 'true'
+            }
+            steps {
+              script {
+                env.CEPH_BUILDER_IMAGE = "${env.CONTAINER_REPO_HOSTNAME}/${env.CONTAINER_REPO_ORGANIZATION}/ceph-build"
+                def ceph_builder_tag = "${env.BRANCH}.${env.DIST}.${ARCH}"
+                sh '''#!/bin/bash
+                  set -ex
+                  podman login -u ${CONTAINER_REPO_CREDS_USR} -p ${CONTAINER_REPO_CREDS_PSW} ${CONTAINER_REPO_HOSTNAME}/${CONTAINER_REPO_ORGANIZATION}
+                '''
+                def os = get_os_info(env.DIST)
+                sh """#!/bin/bash
+                  set -ex
+                  cd dist/ceph
+                  sed -i'' -e 's/\\(apt-get install .*\\)/\\1 curl/' ./src/script/buildcontainer-setup.sh
+                  python3 src/script/build-with-container.py --image-repo=${env.CEPH_BUILDER_IMAGE} --tag=${ceph_builder_tag} -d ${DIST} -e build-container
```

Comment:
The `build-container` step always builds the image. Do you want it always rebuilt in this step?

Reply:
Well I'm not sure; I did this initially because otherwise, it would never get re-built if it had ever been built before.

Reply:
fair enough. this is certainly safe for now and we can revisit this part in the future when needed.

Reply:
I do want to experiment with that; there's just probably a dozen things ahead of it in line